src/blame_git.c

```diff
@@ -365,11 +365,14 @@ static void trim_common_tail(mmfile_t *a, mmfile_t *b, long ctx)
 	b->size -= trimmed - recovered;
 }
 
-static int diff_hunks(mmfile_t file_a, mmfile_t file_b, void *cb_data)
+static int diff_hunks(mmfile_t file_a, mmfile_t file_b, void *cb_data, git_blame_options *options)
 {
-	xpparam_t xpp = {0};
 	xdemitconf_t xecfg = {0};
 	xdemitcb_t ecb = {0};
+	xpparam_t xpp = {0 };
+	if (options->flags & GIT_BLAME_IGNORE_WHITESPACE) {
+		xpp.flags |= XDF_IGNORE_WHITESPACE;
+	}
```

Comment:
Small style nit: we usually avoid braces around single-line statements in conditions. Please also remove the space in `{0 }` and add a newline between variables and first statement:

```
xpparam_t xpp = {0};

if (options->flags & GIT_BLAME_IGNORE_WHITESPACE)
        xpp.flags |= XDF_IGNORE_WHITESPACE;
```

Reply:
Done and I rebased all my commits into one like you recommended me in IRC.